apps/services/personal-representative-public/src/app/modules/personalRepresentatives/personalRepresentatives.controller.ts

```diff
@@ -56,36 +58,37 @@ export class PersonalRepresentativeRightsController {
       'Gets personal representative rights by nationalId of personal representative',
     description: 'A personal representative can represent more than one person',
   })
-  @Get(':nationalId')
+  @Get()
+  @ApiQuery({
+    name: 'prId',
+    required: true,
+    type: 'string',
+    description: 'nationalId of personal representative.',
+  })
   @ApiOkResponse({
     description: 'Personal representative connections with rights',
-    type: PersonalRepresentativePublicDTO,
-  })
-  @ApiParam({ name: 'nationalId', required: true, type: String })
-  @Audit<PersonalRepresentativePublicDTO[]>({
-    resources: (prRights) =>
-      prRights.map((pr) => pr.nationalIdRepresentedPerson),
+    type: [PersonalRepresentativePublicDTO],
   })
   async getByPersonalRepresentative(
-    @Param('nationalId') nationalId: string,
+    @Query('prId') prId: string,
     @CurrentAuth() user: Auth,
   ): Promise<PersonalRepresentativePublicDTO[]> {
-    if (!nationalId) {
-      throw new BadRequestException('NationalId needs to be provided')
+    if (!prId) {
+      throw new BadRequestException(
+        'NationalId of personal representative needs to be provided',
+      )
     }
 
     const personalReps = await this.auditService.auditPromise(
       {
-        user,
+        auth: user,
         action: 'getPersonalRepresentativePermissions',
         namespace,
-        resources: nationalId,
+        resources: prId,
```

Comment:
I don't think there is a real reason for that since most cases are 0-1 represented with very rare instances of maybe 2-4 where a single personal rep is representing more than 1 person.
In that case you can get the length of the array for count and having pagination would be a waste of effort and unnecessary complexity